frontend/src/App.js

```diff
@@ -1,9 +1,284 @@
-import React from 'react'
+import React, { useState } from "react";
+import styled from "styled-components";
 
 export const App = () => {
+  const [signupPage, setSignupPage] = useState(false);
+  const [loginPage, setLoginPage] = useState(false);
+  const [welcomePage, setWelcomePage] = useState(false);
+  const [welcomeMessage, setWelcomeMessage] = useState();
+  const [errorMessage, setErrorMessage] = useState("");
+  const [name, setName] = useState("");
+  const [password, setPassword] = useState("");
+
+  const SIGNUP_URL = "https://auth-project-annika-caroline.herokuapp.com/users";
+  const LOGIN_URL = "https://auth-project-annika-caroline.herokuapp.com/sessions";
+  const WELCOME_URL = "https://auth-project-annika-caroline.herokuapp.com/welcome";
+
+  const signupUser = (event) => {
+    event.preventDefault();
+    setErrorMessage("")
+    fetch(SIGNUP_URL, {
+      method: "POST",
+      headers: { "Content-Type": "application/json" },
+      body: JSON.stringify({ name, password }),
+    })
+      .then((res) => {
+        if (!res.ok) {
+          throw new Error(
+            "Please try another username."
+          );
+        } else {
+          return res.json();
+        }
+      })
+      .then((json) => {
+        welcomeUser(json.accessToken);
+      })
+      .catch((err) => {
+        setErrorMessage(err.message);
+      })
+      .finally(() => {
+        setName("");
+        setPassword("");
```

Comment:
You need to store your token in the local storage to make your users still logged in after refreshing the page.